### DISCUSSION TITLE
access to summary_variables

### DISCUSSION THREAD

**Discussion by masoodtamaddon** [pybamm-team/PyBaMM#1671] [Q&A] [answered]
I have tried to run Calendar aging example and include the summary_varibales in it to track the capacity fade, but it seems like these variables are inaccessible. Please see the following script. 
(In fact, I have only managed to call these variables in the long_experiment script and nowhere else).  

import pybamm as pb
import matplotlib.pyplot as plt
import numpy as np

pb.set_logging_level("INFO")

model= pb.lithium_ion.DFN({"SEI": "reaction limited"})
param = model.param
parameter_values = model.default_parameter_values
parameter_values["Current function [A]"] = 0

##Calculate the stoic at Cell SOC, Enforce the initial concentration attributed to the SOC level
x_init, y_init = pb.lithium_ion.get_initial_stoichiometries(initial_soc=1 , parameter_values=parameter_values)
print(f"Initial negative electrode SOC: {x_init:.3f}")
print(f"Initial positive electrode SOC: {y_init:.3f}")
c_n_max = parameter_values.evaluate(param.c_n_max)
c_p_max = parameter_values.evaluate(param.c_p_max)
parameter_values.update(
    {
        "Initial concentration in negative electrode [mol.m-3]": x_init * c_n_max,
        "Initial concentration in positive electrode [mol.m-3]": y_init * c_p_max,
    }
)

sim = pb.Simulation(model, parameter_values=parameter_values)
solver = pb.CasadiSolver(mode="fast")

years = 30
years_secs = years * 365 * 24 * 3600
t_eval = np.linspace(0, years_secs, 100)

sol = sim.solve(t_eval=t_eval, solver=solver)

pb.dynamic_plot(
    sim,
    [
        "Terminal voltage [V]",
        "Negative particle surface concentration",
        "X-averaged negative particle surface concentration",
        "Electrolyte concentration [mol.m-3]",
        "Total negative electrode SEI thickness [m]",
        "Loss of lithium inventory [%]",
    ],
)

fig, ax = plt.subplots()
ax.plot(sol.summary_variables["Capacity [A.h]"])
fig.tight_layout()

**Comment by valentinsulzer** [comment, marked as answer]
Which version are you using? Can you update to latest version?

**Reply by masoodtamaddon** [reply]
It's release v21.08. Other features of this release (e.g. eSOH and get_initial_stoichiometries) are working completely alright.

**Reply by valentinsulzer** [reply]
Right, I see the problem, summary variables are fundamentally tied to experiments (you need the concept of a "cycle"), so they can't be calculated if you don't use the experiment. See https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/models/electrode-state-of-health.ipynb if you want to calculate capacity at the end of your simulation (you could do it at each timestep too if you want)
